dashboard/src/actions/comparisonActions.js

```diff
@@ -0,0 +1,122 @@
+import * as TYPES from "./types.js";
+
+import { DANGER, ERROR_MSG } from "assets/constants/toastConstants";
+
+import API from "../utils/axiosInstance";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "../utils/helper";
+
+const chartTitleMap = {
+  gb_sec: "Bandwidth",
+  trans_sec: "Transactions/second",
+  usec: "Latency",
+};
+
+export const getQuisbyData = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    dispatch({
+      type: TYPES.SET_ACTIVE_RESOURCEID,
+      payload: dataset.resource_id,
+    });
+    const response = await API.get(
+      uriTemplate(endpoints, "datasets_visualize", {
+        dataset: dataset.resource_id,
+      })
+    );
+    if (response.status === 200 && response.data.json_data) {
+      dispatch({
+        type: TYPES.SET_QUISBY_DATA,
+        payload: response.data.json_data,
+      });
+      dispatch({
+        type: TYPES.IS_UNSUPPORTED_TYPE,
+        payload: "",
+      });
+      dispatch(parseChartData());
+    }
+  } catch (error) {
+    if (error?.response && error.response?.data) {
```

Comment:
This isn't important, but I think `error?.response?.data` would have exactly the same result as `error?.response && error.response?.data` 😄